_posts/2021-01-25-solidity-developer-survey-2020-results.md

```diff
@@ -0,0 +1,358 @@
+
+Massive open online courses (MOOCs) like Udacity, Udemy and university courses were mentioned by several respondents followed by active coding approaches like "learning by coding" using Remix and the OpenZeppelin contracts as well as by going through the ConsenSys Academy. 
+
+![Primary Learning Resources](/img/2021/01/PrimaryResource.png)
+
+#### Problem Solving 
+
+When getting stuck on a Solidity problem, most Solidity developers visit Ethereum StackExchange for help. 
+
+Other popular solutions are asking friends or coworkers for help, watching tutorial videos or doing other work and trying to solve the problem later.
+
+![Problem Solving](/img/2021/01/SolidityProblem.png)
+
+When asked if developers encounter similar issues repeatedly, most responded with "no" or explained that with experience and usage, once they knew how to solve the problem they did not encounter the same issues repeatedly.
+
+The following issues have been mentioned to still pop up regularly or to be annoying:
+- Stack too deep.
+- Error messages that don't point to the cause of the issue.
+- Out of gas errors.
+- Compilation errors.
+- Tooling problems. 
+- Verification difficulties.
+- Breaking changes and broken dependencies.
+
+### (3) Understanding the Solidity Developer Experience
+
+#### Solidity Usage 
+
+Roughly 65% of respondents have been using Solidity for more than a year, with 43% being real Solidity pros with more than 2 years of experience with the language.
+
+It's interesting to see that a fair share of 35% is just getting started with Solidity and has only been using it for under 12 months. 
+
+Looking at the overall coding seniority of the respondents, this data shows us that a majority of Solidity developers are developers with experience in other programming languages, rather than someone learning to program for the first time in Solidity.
+
+![Solidity Usage](/img/2021/01/UsingSol.png)
+
+43% of respondents rate themselves Solidity experts with an expertise level of 8 (out of 10) or higher. Only 10% rank themselves at the beginner stages with an expertise level of 3 or lower.
+
+![Expertise](/img/2021/01/SolExpertise.png)
+
+The majority of respondents use Solidity on a weekly (31%) or even daily basis (41%). Almost 18% use it monthly, while the remaining 9% stated that they rarely code in Solidity.
+
+![Frequency of Usage](/img/2021/01/HowOftenSol.png)
+
+Most respondents are currently using 0.6.x or 0.7.x versions of Solidity. A smaller share still uses 0.5.x versions or has been trying out the recently released 0.8.0 already.
+
+A few people are not aware of the version they are using or are still using very old versions from the 0.4.x series. 
+
+🚨 Please make sure to frequently update your code since [several important bug fixes](https://github.com/ethereum/solidity/blob/develop/docs/bugs_by_version.json#L922) and security improvements have been added since 0.4.x!
```

Comment:
It's pretty interesting that the list of important bugs becomes nearly empty with 0.7.x. It doesn't seem to have anything to do with release frequency (0.5.x had tons of them even with more than one release a month sometimes). Maybe it's the effect of better code review coverage with larger team?